web/pdf_viewer.js

```diff
@@ -76,6 +93,36 @@ class PDFViewer extends BaseViewer {
       location: this._location,
     });
   }
+
+  _regroupSpreads() {
+    requestAnimationFrame(() => {
```

Comment:
Is it ever possible for `_regroupSpreads` to be called twice within one frame?
If not, add a comment why not.
If so, then you need to use a boolean flag to check whether `rAF` has already been called.
Maybe you even need to  save the handle of `requestAnimationFrame`, and cancel the frame request if needed (I can imagine that when you want to start printing, that all pending operations may need to be canceled).

Reply:
I didn't want to overengineer this, since the need for `requestAnimationFrame` at all is somewhat speculative. (‘It's a good idea to do heavy DOM manipulation within `rAF` if you care about performance’ is the entirety of my rationale; I confess to not knowing enough about browser rendering internals to examine that maxim in more detail.) That said, the only way that `_regroupSpreads` gets called is if the spread mode is actually changed from its previous value. So calling it multiple times between frames would mean the user is clicking a spread mode button multiple times between frames. Even then, nothing incorrect happens; there's just a redundant manipulation of the DOM. So the upshot is that the worst thing that can happen is that someone can spam the UI with clicks and get a performance hit as a consequence. To that I say... shrug? But implementing the throttling pattern in an abundance of caution wouldn't be hard either. What do you think?

Reply:
Have you experienced any performance issues without rAF? If not, just remove it.

rAF should be used to avoid sequences of repeated layout trashing, and from a quick glance there doesn't seem to be any obvious signs of that in the code (maybe via callers of `setInitialView`? I didn't really follow the call sites of that).